pkg/synopsysctl/cmd_migrate.go

```diff
@@ -324,24 +348,127 @@ func migrateBlackDuck(namespace string) error {
 		}
 
 		if blackDuck.Spec.PersistentStorage {
-			err = removePVC(blackDuckNamespace)
+			var rabbitmqRCName, zookeeperRCName, uploadCacheRCName, uploadCacheKeyPVCName, uploadCacheDataPVCName string
+			if value, ok := blackDuck.Annotations["synopsys.com/created.by"]; ok && "pre-2019.6.0" == value {
+				rabbitmqRCName = "rabbitmq"
+				zookeeperRCName = "zookeeper"
+				uploadCacheRCName = "uploadcache"
+				uploadCacheKeyPVCName = "blackduck-uploadcache-key"
+				uploadCacheDataPVCName = "blackduck-uploadcache-data"
+			} else {
+				rabbitmqRCName = util.GetResourceName(blackDuckName, util.BlackDuckName, "rabbitmq")
+				zookeeperRCName = util.GetResourceName(blackDuckName, util.BlackDuckName, "zookeeper")
+				uploadCacheRCName = util.GetResourceName(blackDuckName, util.BlackDuckName, "uploadcache")
+				uploadCacheKeyPVCName = fmt.Sprintf("%s-blackduck-uploadcache-key", blackDuckName)
+				uploadCacheDataPVCName = fmt.Sprintf("%s-blackduck-uploadcache-data", blackDuckName)
+			}
+			// scale down zookeeper
+			err = scaleDownRC(blackDuckNamespace, zookeeperRCName)
+			if err != nil {
+				return err
+			}
+			// scale down upload cache
+			err = scaleDownRC(blackDuckNamespace, uploadCacheRCName)
+			if err != nil {
+				return err
+			}
+			err = migrateUploadCachePVCJob(blackDuckNamespace, blackDuckName, uploadCacheKeyPVCName, uploadCacheDataPVCName)
 			if err != nil {
 				return err
 			}
+
+			pvcs := []string{"blackduck-rabbitmq"}
+			if value, ok := blackDuck.Annotations["synopsys.com/created.by"]; ok && "pre-2019.6.0" == value {
+				pvcs = append(pvcs, "zookeeper-data", "zookeeper-datalog")
+			} else {
+				pvcs = append(pvcs, util.GetResourceName(blackDuckName, util.BlackDuckName, "zookeeper-data"), util.GetResourceName(blackDuckName, util.BlackDuckName, "zookeeper-datalog"))
+			}
+			for _, pvc := range pvcs {
+				err = removePVC(blackDuckNamespace, pvc, rabbitmqRCName)
+				if err != nil {
+					return err
+				}
+			}
 		}
 		log.Infof("successfully migrated Black Duck '%s' in namespace '%s'", blackDuckName, blackDuckNamespace)
 	}
 	return nil
 }
 
-func removePVC(blackDuckNamespace string) error {
-	_, err := util.GetPVC(kubeClient, blackDuckNamespace, "blackduck-rabbitmq")
+// removePVC removes the PVC
+func removePVC(namespace string, name string, rabbitmqRCName string) error {
+	_, err := util.GetPVC(kubeClient, namespace, name)
 	if err == nil {
-		return util.DeletePVC(kubeClient, blackDuckNamespace, "blackduck-rabbitmq")
+		if "blackduck-rabbitmq" == name {
```

Comment:
Move this check and scale down of rabbitmq to outside the function.  removePVC should only be responsible for removing the PVC told to remove.

Reply:
done